openedx/features/subscriptions/api/v1/views.py

```diff
@@ -0,0 +1,58 @@
+"""
+HTTP endpoints for interacting with subscriptions.
+"""
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.generics import ListAPIView, RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from django.http import Http404
+
+from lms.djangoapps.commerce.api.v1.permissions import ApiKeyOrModelPermission
+from openedx.core.lib.api.mixins import PutAsCreateMixin
+from openedx.features.subscriptions.models import UserSubscription
+from openedx.features.subscriptions.api.v1.serializers import UserSubscriptionSerializer
+
+
+class SubscriptionsListView(ListAPIView):
+    """
+    List subscriptions.
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserSubscriptionSerializer
+    pagination_class = None
+
+    def get_queryset(self):
+        return_only_valid_subscriptions = True if self.request.query_params.get('valid') == 'true' else False
+        user_id = self.request.query_params.get('user', self.request.user.id)
+        if return_only_valid_subscriptions:
+            return UserSubscription.get_valid_subscription(user_id=user_id)
+
+        return UserSubscription.objects.filter(user_id=user_id).order_by('-created')
```

Comment:
Add ordering on class level, just like we did for `SubscriptionRetrieveUpdateView` (https://www.django-rest-framework.org/api-guide/filtering/#specifying-which-fields-may-be-ordered-against):
```python
    filter_backends = [filters.OrderingFilter]
    ordering_fields = ['username', 'email']
```

Reply:
Although this is a List view, it would ideally only return 0 or 1 subscriptions for a given user, doesn't make sense to order it in any way.

Reply:
Since this is a list view so I would still prefer the explicit ordering.

Reply:
done. I don't think we need the `OrderingFilter`. I have done it like it was done in the retrieve update view